Exception : unwanted situation that occured during runtime called exception.if exception not handled then next step of the program wont get excecuted
why need to do exception ?
To run our code util end , we need to do exception.

try : that might have exception
catch : that handle exception 
finally{ } : print statement , with or without exception it will do its operation.
throw : throws an exception
throws : method that may contain exception.
checkedException : 
*) ArithematicException 
class Main {
    public static void main(String[] args) {
        int a = 5;

        for (int b = 0; b < 5; b++) {
            try{
                int c = a/b;
                System.out.println("Result of " + a + "/" + b + " = " + c);
        }catch(Exception e){
            System.out.println(e);  // just by saying this much it will handle it ,by catch(Exception e) also handle exception no need of new method.
        }
        }
    }
}
*) NullPointerException 
class Main{
    public static void main(String[] args){
        String str1 = "vinay";
        String str2 = null; //  null value
        String str3 = "hello";
     str1 = str1.toUpperCase();
     System.out.println(str1);
     try{
     str2 = str2.toUpperCase();
     System.out.println(str2);
     }catch(NullPointerException e){
         System.out.println("Null value cannot connverted to uppercase");
     }
     str3 = str3.toUpperCase();
     System.out.println(str3);
    }
} 
output : 
VINAY
Null value cannot connverted to uppercase
HELLO

